### PR TITLE
Use Loofah to parse text from HTML emails for email commands

### PR DESCRIPTION
**Pull request by sampoder**
Fixes https://github.com/hackclub/hcb/issues/5389, problem description in https://hackclub.slack.com/archives/C047Y01MHJQ/p1709574806372209

### CODE REVIEW

File: app/mailboxes/hcb_code_mailbox.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def process
   private
 
   def set_commands
-    content = text || body || html
+    content = text || Loofah.html5_fragment(html).to_text || body
```
Comment (garyhtou) [inline app/mailboxes/hcb_code_mailbox.rb]: what type of content is `body`? (HTML?, text?, anything?)

Comment (sampoder) [reply app/mailboxes/hcb_code_mailbox.rb]: It's the equivalent to the contents of an `.eml` file afaik. For plain text emails, it's just plain text

**Comment by northeastprince** [comment]
If we're using the library directly now we should put it in the Gemfile. Also, is there no other simple way to do this without it?

**Comment by sampoder** [comment]
This feels fairly simple to me; there was nothing built in that replicated JavaScript's innerText implementation

**Comment by northeastprince** [comment]
I agree the API is great for our use case, but if there's a way to do it with comparable effort without relying on another dependency, I would [heavily prefer that](https://www.jorgemanrubia.com/2018/03/15/mind-your-dependencies). That said, it's a utility library which are pretty easy to switch out if needed, so it's not a huge deal.

**Review by garyhtou — COMMENTED** [review]
All good with using Loofah!

The merge wand is yours 🪄! Please test & merge!

**Pull request merged**